feedback/models.py

```diff
@@ -344,6 +417,12 @@ class Meta:
     path_key = models.CharField(max_length=255, db_index=True)
     max_grade = models.PositiveSmallIntegerField(default=MAX_GRADE)
 
+    conversation = models.ForeignKey(
+        Conversation,
+        related_name='feedbacks',
+        on_delete=models.PROTECT,
```

Comment:
Just curious, why did you choose `on_delete PROTECT`? It is not common to delete Feedback (it never happens automatically?) so this does not make a huge impact anywhere. So it is now not allowed to delete conversations until all feedback related to the conversion have been deleted first.

Reply:
Good question. To be honest, the answer is kind of "I wasn't sure, so this seemed like the safest solution." It probably could have also been `CASCADE` as well. Basically the conversation model is to group all existing feedback responses by a certain student to a certain exercise, so as long as there are feedback responses, the conversation model should exist. I wasn't sure how easy or difficult it is to delete models, so using `PROTECT` rather than `CASCADE` does minimize the risk of accidentally deleting feedback responses. But I suppose if one were to actually want to delete feedback responses in batches, it would be convenient to delete the `Conversations` and have the deletion cascade.